src/tools/profile/read_profile.c

```diff
@@ -131,49 +167,76 @@ static double time_to_usec(profile_data_t *data, options_t *opts, uint64_t time)
 
 static int compare_locations(const void *l1, const void *l2)
 {
-    const ucs_profile_location_t *loc1 = l1;
-    const ucs_profile_location_t *loc2 = l2;
+    const ucs_profile_thread_location_t *loc1 = l1;
+    const ucs_profile_thread_location_t *loc2 = l2;
     return (loc1->total_time > loc2->total_time) ? -1 :
```

Comment:
return loc2->total_time - loc1->total_time

Reply:
total_time is 64-bit and need to return 32-bit, so it will not work..

Reply:
ok